src/hooks/useActionCenterSubscribers/useArbitrumBridgeActionSubscriber.tsx

```diff
@@ -0,0 +1,211 @@
+
+    allClaims.forEach(claim => {
+      const withdrawTxHash = claim.tx.txid
+
+      const existingAction = Object.values(store.getState().action.byId).find(
+        action =>
+          isArbitrumBridgeWithdrawAction(action) &&
+          action.arbitrumBridgeMetadata.withdrawTxHash === withdrawTxHash,
+      )
+
+      if (existingAction) return
+
+      dispatch(
+        actionSlice.actions.upsertAction({
+          id: uuidv4(),
+          createdAt: claim.tx.blockTime * 1000,
+          updatedAt: Date.now(),
+          type: ActionType.ArbitrumBridgeWithdraw,
+          status: ActionStatus.Initiated,
+          arbitrumBridgeMetadata: {
+            withdrawTxHash,
+            amountCryptoBaseUnit: claim.amountCryptoBaseUnit,
+            assetId: claim.assetId,
+            destinationAssetId: claim.destinationAssetId,
+            destinationAddress: claim.destinationAddress,
+            accountId: claim.accountId,
+            chainId: claim.tx.chainId,
+            destinationChainId: claim.destinationChainId,
+          },
+        }),
+      )
+    })
+  }, [dispatch, claimsByStatus.Pending, claimsByStatus.Available])
+
+  const determineActionState = useCallback(
+    (action: ArbitrumBridgeWithdrawAction, withdrawTxHash: string): ActionUpdate | null => {
+      const availableClaim = claimsByStatus.Available.find(
+        claim => claim.tx.txid === withdrawTxHash,
+      )
+      const completedClaim = claimsByStatus.Complete.find(claim => claim.tx.txid === withdrawTxHash)
+      const pendingClaim = claimsByStatus.Pending.find(claim => claim.tx.txid === withdrawTxHash)
+
+      const currentMetadata = action.arbitrumBridgeMetadata
+
+      const getNewState = () => {
+        if (completedClaim) {
+          return {
+            newStatus: ActionStatus.Claimed,
+            claimDetails: completedClaim,
+            timeRemainingSeconds: currentMetadata.timeRemainingSeconds,
+            claimTxHash: currentMetadata.claimTxHash,
+          }
+        }
+
+        if (availableClaim) {
+          return {
+            newStatus: ActionStatus.ClaimAvailable,
+            claimDetails: availableClaim,
+            timeRemainingSeconds: availableClaim.timeRemainingSeconds,
+            claimTxHash: currentMetadata.claimTxHash,
+          }
+        }
+
+        if (pendingClaim) {
+          return {
+            newStatus: ActionStatus.Initiated,
+            claimDetails: pendingClaim,
+            timeRemainingSeconds: pendingClaim.timeRemainingSeconds,
+            claimTxHash: currentMetadata.claimTxHash,
+          }
+        }
+
+        return {
+          newStatus: action.status,
+          claimDetails: currentMetadata.claimDetails,
+          timeRemainingSeconds: currentMetadata.timeRemainingSeconds,
+          claimTxHash: currentMetadata.claimTxHash,
+        }
+      }
+
+      const newState = getNewState()
+
+      const hasChanges =
```

Comment:
Pretty cool stuff here is that this allows us to be more granular if we want to - say in the future we want to only update if say a full day has elapsed since last time remaining secs - we could absolutely do that